pilot/pkg/model/validation.go

```diff
@@ -1476,19 +1476,49 @@ func ValidateServiceRole(name, namespace string, msg proto.Message) error {
 			if len(constraint.Values) == 0 {
 				errs = appendErrors(errs, fmt.Errorf("at least 1 value must be specified for constraint %d in rule %d", j, i))
 			}
+			if hasExistingFirstClassFieldInRole(constraint.Key, rule) {
+				errs = appendErrors(errs, fmt.Errorf("cannot define %s for rule %d because a similar first-class field has been defined", constraint.Key, i))
+			}
 		}
 	}
 	return errs
 }
 
+// Returns true if the user defines a constraint that already exists in the first-class fields, false
+// if none has overlapped.
+func hasExistingFirstClassFieldInRole(constraintKey string, rule *rbac.AccessRule) bool {
```

Comment:
It's a bit hard to understand what are first class fields. Can you add a comment here to explain what are they, and why only port are need to be checked. May even consider rename the function if it cares about port only.

Reply:
Sounds good. I updated the comment to explain what first-class fields are. Not sure if we should rename the function since we may introduce more first class fields in the future that matches existing non first-class fields. Thanks!

Reply:
This function only check the `destination.port`, so the description is still not very clear to me (i.e why don't you check other first class fields). Do I miss something?